src/main/kotlin/org/codefreak/codefreak/service/evaluation/EvaluationStepProcessor.kt

```diff
@@ -187,7 +187,7 @@ class EvaluationStepProcessor : ItemProcessor<EvaluationStep, EvaluationStep?> {
     val stepDefinition = step.definition
     return DefaultEvaluationRunConfig(
         id = step.id,
-        imageName = appConfig.workspaces.companionImage,
+        imageName = step.definition.task.customWorkspaceImage ?: appConfig.workspaces.companionImage,
```

Comment:
I think we just declared this a few lines above ^^

```suggestion
        imageName = stepDefinition.task.customWorkspaceImage ?: appConfig.workspaces.companionImage,
```